### PR TITLE
Add policy on pull request review and merge

### PR DESCRIPTION
**Pull request by taldcroft**
Per discussion in CoCo, add a new policy that describes the Astropy Project policy on reviewing pull requests and subsequent merge.

### CODE REVIEW

File: policies/pull-request-review-and-merge.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  - New tests or modifications to tests are adequate to cover the code changes.
+  - There are no API changes or the API changes are well-understood and
+    acceptable and beneficial on the whole to the astropy user community.
+  - The expected level of review is detailed, including review and suggestions
+    at the line-by-line level.  Formatting and style are acceptable points for
+    comment, in particular to maintain the existing style of a package and to
+    maintain readability of the code base for future developers.
+  - A maintainer of  sub-packages that the contribution affects is required to
+    approve the pull request for changes that require specific knowledge of a
+    sub-package.  Other less complex cases, for example a documentation
```
Comment (pllim) [inline policies/pull-request-review-and-merge.md:20]:
```suggestion
    subpackage.  Other less complex cases, for example a documentation
```

File: policies/pull-request-review-and-merge.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
+# Policy on pull request review and merging in the Astropy Project
+
+## Astropy core
+
+In the astropy core package the requirements are as follows:
+
+- At least one contributor with relevant expertise must review the pull request
+  in detail.
+  - New or updated code is logically correct and achieves the desired effect of
```
Comment (pllim) [inline policies/pull-request-review-and-merge.md:9]: Should we change the indentation from 2-space to 4-space? At first, I thought the indentation is a typo.

File: policies/pull-request-review-and-merge.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    maintain readability of the code base for future developers.
+  - A maintainer of  sub-packages that the contribution affects is required to
+    approve the pull request for changes that require specific knowledge of a
+    sub-package.  Other less complex cases, for example a documentation
+    improvement, may be reviewed by any astropy core package maintainer.
+  - While a maintainer does not have to be the one doing the detailed review, if
+    the detailed review is by someone else, the maintainer is responsible for
+    ensuring the contributor review is sufficiently detailed and follows these
+    guidelines.
+  - After successful review, maintainers should formally approve the pull
```
Comment (pllim) [inline policies/pull-request-review-and-merge.md:26]:
```suggestion
  - Following a successful review, maintainers should formally approve the pull
```

File: policies/pull-request-review-and-merge.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    request.
+- A release manager or other maintainer checks that the selected release
```
Comment (pllim) [inline policies/pull-request-review-and-merge.md:28]:
```suggestion
- A release manager or maintainer checks that the selected release
```

File: policies/pull-request-review-and-merge.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  milestone is suitable for the scope of the change.  In particular bug-fix
+  backports are considered where feasible and sensible.
```
Comment (pllim) [inline policies/pull-request-review-and-merge.md:30]:
```suggestion
  backports are considered where feasible and sensible.
  Documentation is usually backported. API change or new feature
  are not backported unless under special circumstances.
```

File: policies/pull-request-review-and-merge.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+- A matrix of CI checks ensure that all existing tests pass on supported
+  platforms.
```
Comment (pllim) [inline policies/pull-request-review-and-merge.md:32]:
```suggestion
  platforms. Even the CI job that is allowed to fail should pass unless
  an known and unrelated failure occurs; i.e., look at the logs even in
  the event of green check mark.
```

File: policies/pull-request-review-and-merge.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+- A bot checks that the change log mentions the change and that the changelog
```
Comment (pllim) [inline policies/pull-request-review-and-merge.md:33]:
```suggestion
- A bot checks that the change log mentions the change and PR number and that the change log
```

File: policies/pull-request-review-and-merge.md [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+  section used matches the selected release milestone.
```
Comment (pllim) [inline policies/pull-request-review-and-merge.md:34]:
```suggestion
  section used matches the selected release milestone set by the maintainer.
```

File: policies/pull-request-review-and-merge.md [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+- Merging the pull request can be done by any core package maintainer once
+  approved by relevant maintainers (as described above).  In practice this is
+  sometimes the PR developer and sometimes the reviewer.
+
+## Astropy coordinated and infrastructure packages
+
+The process should in general be similar to the core package, although since a
+number of coordinated and infrastructure packages have fewer maintainers, the
+concept of sub-packages is not relevant. Nevertheless, all coordinated and
```
Comment (pllim) [inline policies/pull-request-review-and-merge.md:43]:
```suggestion
concept of subpackages is not relevant. Nevertheless, all coordinated and
```

File: policies/pull-request-review-and-merge.md [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+infrastructure packages should have at least two maintainers with push access.
+All changes to these packages should be done via pull requests, and approved by
+at least one of the other maintainers (any maintainer can then merge the pull
+request provided it is approved).
+
+## Astropy affiliated packages
+
+The change process required by the Astropy Project for affiliated packages are
+substantially less rigid than in the Astropy core package. However, each
+affiliated package may make their requirements as rigorous as they’d like.
```
Comment (pllim) [inline policies/pull-request-review-and-merge.md:53]: Maybe point to https://github.com/astropy/project/blob/master/affiliated/affiliated_package_review_guidelines.md , because that is the standard they have to uphold if we ever implement re-reviews.

**Comment by pllim** [comment]
Forgot to add, I think it is also important for reviewers and authors to adhere to code of conduct during the review process. That should be stated clearly in the policy with a link to https://github.com/astropy/astropy/blob/master/CODE_OF_CONDUCT.md .

File: policies/pull-request-review-and-merge.md (hunk 2 above)
Comment (taldcroft) [reply policies/pull-request-review-and-merge.md:9]: I think it doesn't matter, but in markdown I always use 2.

File: policies/pull-request-review-and-merge.md (hunk 10 above)
Comment (taldcroft) [reply policies/pull-request-review-and-merge.md:58]: It seems that the package review guidelines do not address the process by which a package is maintained, all that matters (from the perspective of review guidelines) is the end result. So my preference is to not include a link here.

File: policies/pull-request-review-and-merge.md [hunk 11]
```diff
@@ -0,0 +1,65 @@
+# Policy on pull request review and merging in the Astropy Project
+
+## Astropy core
+
+In the astropy core package the requirements are as follows:
+
+- At least one contributor with relevant expertise must review the pull request
+  in detail.
+  - New or updated code is logically correct and achieves the desired effect of
```
Comment (eteq) [inline policies/pull-request-review-and-merge.md:9]:
```suggestion
  - New or updated code is correct and achieves the desired effect of
```
I'm not sure what "logically" meant here, but I don't really think it's needed anyway!

File: policies/pull-request-review-and-merge.md [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+    either fixing a bug or making an enhancement.
+  - New tests or modifications to tests are adequate to cover the code changes.
+    Tests for edge cases exist, if applicable.
```
Comment (eteq) [inline policies/pull-request-review-and-merge.md:12]: Should there be something making this a bit softer like:
```suggestion
    Tests for edge cases exist, if applicable and judged to be important enough.
```
so that it doesn't sound like we are demanding all possible edge cases always be accounted for

File: policies/pull-request-review-and-merge.md [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+  - There are no API changes or the API changes are well-understood and
+    acceptable and beneficial on the whole to the astropy user community.
```
Comment (eteq) [inline policies/pull-request-review-and-merge.md:14]:
```suggestion
    acceptable and beneficial on the whole to the astropy user community, and documented as an API change in the changelog.
```

File: policies/pull-request-review-and-merge.md [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+  - The expected level of review is detailed, including review and suggestions
+    at the line-by-line level.  Formatting and style are acceptable points for
```
Comment (eteq) [inline policies/pull-request-review-and-merge.md:16]:
```suggestion
    at the line-by-line level (although "I looked and had no comments" is a valid answer).  Formatting and style are acceptable points for
```

File: policies/pull-request-review-and-merge.md [hunk 15]
```diff
@@ -0,0 +1,65 @@
...
+- A release manager or maintainer checks that the selected release
+  milestone is suitable for the scope of the change.  In particular bug-fix
+  backports are considered where feasible and sensible.
+  Documentation is usually backported. API change or new feature changes
+  are not backported unless under special circumstances.
+- A matrix of CI checks ensure that all existing tests pass on supported
+  platforms. Even the CI job that is allowed to fail should pass unless
+  an known and unrelated failure occurs; i.e., look at the logs even in
+  the event of green check mark.
+- A bot checks that the change log mentions the change and PR number and that the change log
```
Comment (eteq) [inline policies/pull-request-review-and-merge.md:38]:
```suggestion
- An automated tool checks that the change log mentions the change and PR number and that the change log
```
(since it might be actions instead of the bot...)

File: policies/pull-request-review-and-merge.md [hunk 16]
```diff
@@ -0,0 +1,65 @@
...
+The change process required by the Astropy Project for affiliated packages are
+substantially less rigid than in the Astropy core package. However, each
+affiliated package may make their requirements as rigorous as they’d like.
+- It is required to maintain the code under version control on a publicly
+  available site.
+- In most cases packages are hosted on GitHub or GitLab and updated via pull
+  requests, but this is not a requirement.
+- Independent review is encouraged but is at the discretion of the package
+  maintainer(s).  In particular some packages may be largely developed by a
+  single maintainer.
```
Comment (eteq) [inline policies/pull-request-review-and-merge.md:65]:
```suggestion
  single maintainer.
- For affiated packages with enough contributors, the workflow for the core/coordinated packages is recommended, although not required.
```

**Review by eteq — APPROVED** [review]
A bunch of wordsmithing/clarification items, but overall it looks great!
